### PR TITLE
New Jersey 2023 Individual Income Tax Parameter Updates

### LINKED ISSUE #3740: New Jersey 2023 Individual Income Tax Parameter Updates
According to the Form New Jersey Resident Return NJ-1040, 2023 Version, there're some parameter updates.

Reference: https://www.nj.gov/treasury/taxation/pdf/current/1040i.pdf

### PR DESCRIPTION
**Pull request by hua7450**
Fixes #3740

### CODE REVIEW

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/nj/tax/income/credits/ctc/amount.yaml [hunk 1]
```diff
@@ -1,53 +1,47 @@
...
-    # 2021 form listed to show the absence of the credit.
-    - title: Instructions for 2021 Form NJ-1040
-      href: https://www.state.nj.us/treasury/taxation/pdf/other_forms/tgi-ee/2021/1040i.pdf
-    - title: Instructions for 2022 Form NJ-1040 (line 65)
-      href: https://www.state.nj.us/treasury/taxation/pdf/current/1040i.pdf#page=45
-    # Not yet reflected in the statute, but this law schedules it to double in 2023.
-    - title: P.L. 2023, Chapter 72 (1)(a)
-      href: https://pub.njleg.state.nj.us/Bills/2022/AL23/72_.PDF
+      href: https://law.justia.com/codes/new-jersey/2022/title-54a/section-54a-4-17-1/
+    - title: 2021 NJ-1040 Instructions
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/credits/ctc/amount.yaml:11]: looks like ctc didn't exist in 2021 so let's remove this source

File: policyengine_us/parameters/gov/states/nj/tax/income/credits/eitc/match.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+description: New Jersey matches this percentage of the federal earned income tax credit.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/credits/eitc/match.yaml:1]:
```suggestion
description: New Jersey matches this fraction of the federal earned income tax credit.
```

File: policyengine_us/parameters/gov/states/nj/tax/income/credits/property_tax/age_threshold.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+description: New Jersey provides the property tax credit to filers of this age or older.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/credits/property_tax/age_threshold.yaml:1]:
```suggestion
description: New Jersey limits the property tax credit to filers of this age or older.
```

File: policyengine_us/parameters/gov/states/nj/tax/income/exclusions/retirement/pension/percentage/head_of_household.yaml [hunk 4]
```diff
@@ -2,12 +2,16 @@ description: New Jersey allows the exclusion of this fraction of retirement inco
...
   amount_unit: /1
-  threshold_period: year
+  period: year
   type: single_amount
   label: New Jersey pension/retirement exclusion fraction
   reference:
-    - title: Instructions for Form NJ-1040 (line 64)
+    - title: 2021 New Jersey NJ-1040 Resident Return Line 28a
+      href: https://www.state.nj.us/treasury/taxation/pdf/other_forms/tgi-ee/2021/1040i.pdf#page=21
+    - title: 2023 NJ-1040 Instructions Line 28a
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/nj/tax/income/exclusions/retirement/pension/percentage/head_of_household.yaml:11]: move 2022 reference above this

File: policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py [hunk 5]
```diff
@@ -23,21 +26,27 @@ def formula(tax_unit, period, parameters):
         disabled_head = tax_unit("disabled_head", period)
         blind_spouse = tax_unit("blind_spouse", period)
         disabled_spouse = tax_unit("disabled_spouse", period)
-        senior_head = tax_unit("age_head", period) >= p.senior_qualifying_age
-        senior_spouse = (
-            tax_unit("age_spouse", period) >= p.senior_qualifying_age
+        # senior_head = tax_unit("age_head", period) >= p.age_threshold
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nj/tax/income/credits/property_tax_credit/nj_property_tax_credit_eligible.py:29]: delete commented lines

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**